analysis/analysis_test.go

```diff
@@ -92,33 +91,28 @@ func TestApplicationAnalysis(t *testing.T) {
 			// Prepare and submit the analyze task.
 			// tc.Task.Addon = analyzerAddon
 			tc.Task.Application = &api.Ref{ID: tc.Application.ID}
-			taskData := tc.Task.Data.(addon.Data)
+			taskData := tc.Task.Data
 			//for _, r := range tc.CustomRules {
 			//	taskData.Rules = append(taskData.Rules, api.Ref{ID: r.ID, Name: r.Name})
 			//}
-			if len(tc.Sources) > 0 {
-				taskData.Sources = tc.Sources
-			}
-			if len(tc.Targets) > 0 {
-				taskData.Targets = tc.Targets
-			}
-			if len(tc.Labels.Included) > 0 || len(tc.Labels.Excluded) > 0 {
-				taskData.Rules.Labels = tc.Labels
-			}
-			if tc.Rules.Path != "" { // TODO: better rules handling
-				taskData.Rules = tc.Rules
+			if tc.Labels != nil {
+				taskData["rules"] = api.Map{"labels": tc.Labels}
 			}
+			//if tc.Rules["path"] != "" { // TODO: better rules handling
+			//	taskData["rules"] = tc.Rules
+			//}
 			if tc.WithDeps == true {
-				taskData.Mode.WithDeps = true
+				taskData["mode"] = api.Map{"withDeps": true}
 			}
 			if tc.Binary {
-				taskData.Mode.Binary = true
+				taskData["mode"].(api.Map)["binary"] = true
```

Comment:
The blind case causes a panic because the Task.Data is re-used between testcases. As a result the Data.Mode is a map[string]any which cannot be asserted to api.Map.